tracer/test/Datadog.Trace.TestHelpers/ProfilerHelper.cs

```diff
@@ -30,6 +36,61 @@ public class ProfilerHelper
             // clear all relevant environment variables to start with a clean slate
             EnvironmentHelper.ClearProfilerEnvironmentVariables();
 
+            // this is nasty, but it's the only way I could find to force
+            // a .NET Framework exe to run in 32 bit if required
+            if (EnvironmentTools.IsWindows()
+             && !EnvironmentHelper.IsCoreClr()
+             && !EnvironmentTools.IsTestTarget64BitProcess()
+             && Path.GetExtension(executable) == ".exe"
+             && !ExesToExcludeFromCorflags.Contains(Path.GetFileNameWithoutExtension(executable)))
+            {
+                string corFlagsExe = null;
+                var programFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86);
+                var dotnetWindowsSdkToolsFolder = Path.Combine(programFiles, "Microsoft SDKs", "Windows", "v10.0A", "bin");
+
+                if (Directory.Exists(dotnetWindowsSdkToolsFolder))
+                {
+                    // get sub directories, e.g.
+                    // @"C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8.1 Tools",
+                    // @"C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools",
+                    foreach (var folder in Directory.EnumerateDirectories(dotnetWindowsSdkToolsFolder))
+                    {
+                        var exe = Path.Combine(folder, "x64", "CorFlags.exe");
+                        if (File.Exists(exe))
+                        {
+                            corFlagsExe = exe;
+                            break;
+                        }
+                    }
+                }
+
+                if (!string.IsNullOrEmpty(corFlagsExe))
+                {
+                    agent.Output?.WriteLine($"CorFlags.exe found at {corFlagsExe}");
+                    var setBit = EnvironmentTools.IsTestTarget64BitProcess()
```

Comment:
This will always return `false` and `/32BITREQ+` because there's a check for `!EnvironmentTools.IsTestTarget64BitProcess()` above.

Reply:
See Zach's comment 🙂  https://github.com/DataDog/dd-trace-dotnet/pull/3632#discussion_r1063676801